### PR TITLE
Feature/MS-477/Extend highlighter plugin for supporting multi-color highlighting

### PR DESCRIPTION
**Pull request by andrey-ponamarev**
**Related to task:** https://oat-sa.atlassian.net/browse/MS-477
- https://github.com/oat-sa/extension-lti-outcomeui/pull/28
- https://github.com/oat-sa/tao-manual-scoring-fe/pull/159
**Description:** Added support for multicolor highlighting.
**How to test:** Please check related [PR](https://github.com/oat-sa/extension-lti-outcomeui/pull/28)

**Unit tests cli:** `npm run test`

### CODE REVIEW

File: src/highlighter.js [hunk 1]
```diff
@@ -220,6 +260,110 @@ export default function(options) {
...
+        Input:
+                                     ___________
+                                    |           |
+        <span class="yellow"> Lorem |ipsum dolor| sit</span>
+                                    |___________|
+
+        Output:
+         <span class="yellow"> Lorem </span><span class="red">ipsum dolor</span><span class="yellow"> sit</span>
+         */
+        if (isSelectionCoversNodeStart && isSelectionCoversNodeEnd){
```
Comment (AnastasiaArcadia) [inline src/highlighter.js:330]:
```suggestion
        if (isSelectionCoversNodeStart && isSelectionCoversNodeEnd) {
```

File: src/highlighter.js [hunk 2]
```diff
@@ -220,6 +260,110 @@ export default function(options) {
...
+           textNode.parentNode.className = activeClass;
+        } else if (isSelectionCoversNodeStart) {
+           textNode.splitText(selectedRange.endOffset)
```
Comment (AnastasiaArcadia) [inline src/highlighter.js:333]:
```suggestion
           textNode.splitText(selectedRange.endOffset);
```

File: src/highlighter.js [hunk 3]
```diff
@@ -380,12 +525,14 @@ export default function(options) {
                 while (currentNode) {
                     if (isWrappingNode(currentNode)) {
                         inlineRange = {
-                            groupId: currentNode.getAttribute(GROUP_ATTR)
+                            groupId: currentNode.getAttribute(GROUP_ATTR),
+                            c: getColorByClassName(currentNode.className)
```
Comment (AnastasiaArcadia) [inline src/highlighter.js:529]: Will the name like `color` be clearer?

Comment (andrey-ponamarev) [reply src/highlighter.js:529]: Agree, but it was done to save space in DB

**Review by AnastasiaArcadia — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request's target is not `master`

**Review by anshul119 — COMMENTED** [review]
Couldn't test because my env is broken :( Code only
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request's target is not `master`

File: src/highlighter.js [hunk 4]
```diff
@@ -502,15 +693,30 @@ export default function(options) {
         return closestBlackListed.length > 0;
     }
 
+    /**
+     * Wraps text node to the highlighter wrapper element
+     * @param {Node} textNode Text node to wrap
+     * @param {string} className Wrapper class name
+     * @param {number} groupId Group id
+     */
+    function wrap (textNode, className, groupId) {
```
[the remark this answers is not in the document]
Comment (andrey-ponamarev) [reply src/highlighter.js]: Renamed

**Pull request merged**